_playwright-tests/tests/API/Repositories.spec.ts

```diff
@@ -590,4 +594,85 @@ test.describe('Repositories', () => {
       }
     });
   });
+
+  test('Manually trigger repository snapshot', async ({ client, cleanup }) => {
+    const repoName = `manual-snapshot-${randomName()}`;
+    const repoUrl = randomUrl();
+
+    await cleanup.runAndAdd(() => cleanupRepositories(client, repoName, repoUrl));
+
+    let repo: ApiRepositoryResponse;
+    await test.step('Create a repo with snapshot enabled', async () => {
+      repo = await new RepositoriesApi(client).createRepository({
+        apiRepositoryRequest: {
+          name: repoName,
+          url: repoUrl,
+          snapshot: true,
+        },
+      });
+      expect(repo.name).toBe(repoName);
+      expect(repo.url).toBe(repoUrl);
+      expect(repo.snapshot).toBe(true);
+    });
+
+    await test.step('Wait for repository to be valid', async () => {
+      const getRepository = () =>
+        new RepositoriesApi(client).getRepository(<GetRepositoryRequest>{
+          uuid: repo.uuid?.toString(),
+        });
+      const waitWhilePending = (resp: ApiRepositoryResponse) => resp.status === 'Pending';
+      const resp = await poll(getRepository, waitWhilePending, 10);
+      expect(resp.status).toBe('Valid');
+      repo = resp;
+    });
+
+    await test.step('Wait for first snapshot to complete', async () => {
```

Comment:
i don't think you'd need this step since you're waiting for the repository status to be valid. you might also add this line `expect(resp.lastSnapshotUuid).toBeDefined();` to the prior step